### DISCUSSION TITLE
Anyone use Notmuch?

### DISCUSSION THREAD

**Discussion by trev-dev** [meow-edit/meow#286] [Q&A] [answered]
Notmuch and meow's motion-mode don't work well together. Not much has its own way of binding things.

I have gotten around this by defaulting to "insert-mode" on notmuch buffers, but I'm curious: has anyone here already figured out how to get notmuch's keybinds to work well with Meow?

**Comment by Linerre** [comment]
Would defaulting to "normal mode" on notmuch buffers be more convenient? I found myself preferring "normal-mode" wherever the motion mode is the default and a bit awkward. Thus, even in read-only buffers such as *Help*, I have made normal-mode the default, though entering insert-mode makes no sense.

**Reply by trev-dev** [reply]
If you follow the directions for how to configure notmuch, the keybindings for things like "jumping to saved searches" do not work in motion, or normal mode. They do work in inbox mode. I suppose I could re-think/bind my saved searches, but one of the benefits of Meow is usually not having to do that :)

Cc: @DogLooksGood

**Comment by DogLooksGood** [comment, marked as answer]
I recommend never to use insert as the default. It should be either normal or motion. 

If it has text input, use normal, otherwise use motion.

**Reply by trev-dev** [reply]
Hokay, so notmuch binds it's "jump-search" feature to "j" which is not being called by pressing j, or SPC-j. We had spoken once about getting `k` to work in magit with SPC-k from motion mode. How do we get the same with notmuch?

**Reply by trev-dev** [reply]
I got it. Simply switching to motion-mode didn't do the trick. I set the initial state ahead of time and now SPC-j and SPC-k work properly.
```elisp
(dolist (state '((notmuch-hello-mode . motion)
                   (notmuch-search-mode . motion)
                   (notmuch-tree-mode . motion)
                   (notmuch-show-mode . motion)))
    (add-to-list 'meow-mode-state-list state))
```
